services/arrow-flight/src/test/java/com/dremio/service/flight/TestFlightSqlServerWithBasicAuth.java

```diff
@@ -51,4 +55,74 @@ public FlightInfo getFlightInfo(String query) throws SQLException {
     return preparedStatement.execute();
   }
+
+  @Test
+  public void testGetTablesWithoutFiltering() {
+    FlightSqlClient flightSqlClient = getFlightClientWrapper().getSqlClient();
+    FlightInfo flightInfo = flightSqlClient.getTables(null, null, null,
+      null, false);
+    try (FlightStream stream = flightSqlClient.getStream(flightInfo.getEndpoints().get(0).getTicket())) {
+      Assert.assertTrue(stream.next());
+      VectorSchemaRoot root = stream.getRoot();
+      Assert.assertTrue(root.getRowCount() > 0);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Test
+  public void testGetTablesFilteringByCatalogPattern() {
+    FlightSqlClient flightSqlClient = getFlightClientWrapper().getSqlClient();
+    FlightInfo flightInfo = flightSqlClient.getTables("DREMIO", null, null,
+      null, false);
+    try (FlightStream stream = flightSqlClient.getStream(flightInfo.getEndpoints().get(0).getTicket())) {
+      Assert.assertTrue(stream.next());
+      VectorSchemaRoot root = stream.getRoot();
+      Assert.assertTrue(root.getRowCount() > 0);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Test
+  public void testGetTablesFilteringBySchemaPattern() {
+    FlightSqlClient flightSqlClient = getFlightClientWrapper().getSqlClient();
+    FlightInfo flightInfo = flightSqlClient.getTables(null, "INFORMATION_SCHEMA", null,
+      null, false);
+    try (FlightStream stream = flightSqlClient.getStream(flightInfo.getEndpoints().get(0).getTicket())) {
+      Assert.assertTrue(stream.next());
+      VectorSchemaRoot root = stream.getRoot();
+      Assert.assertTrue(root.getRowCount() > 0);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Test
+  public void testGetTablesFilteringByTablePattern() {
+    FlightSqlClient flightSqlClient = getFlightClientWrapper().getSqlClient();
+    FlightInfo flightInfo = flightSqlClient.getTables(null, null, "COLUMNS",
+      null, false);
+    try (FlightStream stream = flightSqlClient.getStream(flightInfo.getEndpoints().get(0).getTicket())) {
+      Assert.assertTrue(stream.next());
+      VectorSchemaRoot root = stream.getRoot();
+      Assert.assertTrue(root.getRowCount() > 0);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Test
+  public void testGetTablesFilteringByTableTypePattern() {
+    FlightSqlClient flightSqlClient = getFlightClientWrapper().getSqlClient();
+    FlightInfo flightInfo = flightSqlClient.getTables(null, null, null,
+      Collections.singletonList("SYSTEM_TABLE"), false);
+    try (FlightStream stream = flightSqlClient.getStream(flightInfo.getEndpoints().get(0).getTicket())) {
+      Assert.assertTrue(stream.next());
+      VectorSchemaRoot root = stream.getRoot();
+      Assert.assertTrue(root.getRowCount() > 0);
+    } catch (Exception e) {
```

Comment:
Instead of catching and rethrowing as a RuntimeException, just change the test method to add throws Exception

Reply:
Fixed